libs/dh/metering-point/feature-upload-measurements/src/upload-page.ts

```diff
@@ -230,9 +232,27 @@ export class DhUploadMeasurementsPage {
     const metadata = this.metadata();
     assertIsDefined(csv);
     assertIsDefined(metadata);
-    this.measurements.send(this.meteringPointId(), metadata.type, csv);
+
+    // Map metering point measure unit to send measurements unit
+    const measurementUnit = this.mapMeasurementUnit(metadata.measureUnit);
+
+    this.measurements.send(this.meteringPointId(), metadata.type, measurementUnit, csv);
   };
 
+  private mapMeasurementUnit(
+    measureUnit: MeteringPointMeasureUnit
+  ): SendMeasurementsMeasurementUnit {
+    switch (measureUnit) {
+      case MeteringPointMeasureUnit.KvArh:
+        return SendMeasurementsMeasurementUnit.KiloVoltAmpereReactiveHour;
+      case MeteringPointMeasureUnit.KWh:
+        return SendMeasurementsMeasurementUnit.KilowattHour;
+      default:
+        // Default to kWh for any other unit types
+        return SendMeasurementsMeasurementUnit.KilowattHour;
+    }
+  }
```

Comment:
suggestion: move this into the `upload-service` instead along with other "mapping" functions.

Reply:
fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/4680/commits/5a9e8e5b18b7662dd4b4f7d587e6c88811756cae